Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityArticulatedHand.cs

```diff
@@ -348,6 +349,56 @@ private void UpdateIndexFingerData(InteractionSourceState interactionSourceState
 #endif // WINDOWS_UWP
         }
 
+#if WINDOWS_UWP
+        // Velocity internal states
+        private float deltaTimeStart;
+        private readonly int velocityUpdateInterval = 6;
+        private int frameOn = 0;
+        private Vector3[] velocityPositionsCache = new Vector3[6];
+        private Vector3[] velocityNormalsCache = new Vector3[6];
+        private Vector3 velocityPositionsSum = Vector3.zero;
+        private Vector3 velocityNormalsSum = Vector3.zero;
+#endif // WINDOWS_UWP
+
+        protected void UpdateVelocity()
+        {
+#if WINDOWS_UWP
+            if (frameOn < velocityUpdateInterval)
+            {
+                velocityPositionsCache[frameOn] = unityJointPositions[(int)HandJointKind.Palm];
+                velocityPositionsSum += velocityPositionsCache[frameOn];
+                velocityNormalsCache[frameOn] = unityJointOrientations[(int)HandJointKind.Palm] * Vector3.up;
+                velocityNormalsSum += velocityNormalsCache[frameOn];
+            }
+            else
+            {
+                int frameIndex = frameOn % velocityUpdateInterval;
+
+                float deltaTime = Time.unscaledTime - deltaTimeStart;
+
+                Vector3 newPosition = unityJointPositions[(int)HandJointKind.Palm];
+                Vector3 newNormal = unityJointOrientations[(int)HandJointKind.Palm] * Vector3.up;
+
+                Vector3 newPositionsSum = velocityPositionsSum - velocityPositionsCache[frameIndex] + newPosition;
+                Vector3 newNormalsSum = velocityNormalsSum - velocityNormalsCache[frameIndex] + newNormal;
+
+                Velocity = (newPositionsSum - velocityPositionsSum) / deltaTime / velocityUpdateInterval;
```

Comment:
Have you considered using an exponential moving average instead of the uniform weight for each frame? It might give better results for sharply changing velocities. Not sure how noticeable that would be in practice but i think it's worth trying out and shouldn't be too hard to implement:
[https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average](https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average)

Reply:
I did consider it. It will become more accurate if we will be taking more frames into account (15+ probably). For small sequences like 6 I'm using (it's taken semi-arbitrarily, will get an average velocity over 100ms time when running at 60 fps, which seemed reasonable), perceived difference will be small. 
Also on a small measurement size, uniform weights should be less susceptible to jitters in hand positions, which we still have.

Reply:
Doesn't matter now as I'll change it to platform-supplied values :)